src/ceph-volume/ceph_volume/util/device.py

```diff
@@ -490,13 +510,21 @@ def _check_lvm_reject_reasons(self):
             # only check generic if no vgs are present. Vgs might hold lvs and
             # that might cause 'locked' to trigger
             rejected.extend(self._check_generic_reject_reasons())
+        if not self.is_acceptable_device_generic:
+            rejected.append('Device type is not acceptable. It should be raw device or partition')
 
         return len(rejected) == 0, rejected
 
-    def _check_raw_reject_reasons(self):
+    def _check_raw_reject_reasons(self, compat):
         rejected = self._check_generic_reject_reasons()
         if len(self.vgs) > 0:
             rejected.append('LVM detected')
+        if comat:
+            if not self.is_acceptable_device_generic:
+                rejected.append('Device type is not acceptable. It should be raw device or partition')
+        else:
+            if not self.is_acceptable_device_raw:
+                rejected.append('Device type is not acceptable. It should be raw device, partition, or crypt')
```

Comment:
Not sure what the new argument is for, but simply use `is_acceptable_device_raw` here to determine if the device works for the raw mode.